static-site/package.json

```diff
@@ -12,6 +12,7 @@
     "test": "gatsby build",
     "build:pp": "gatsby build --prefix-paths",
     "clean": "rm -rf public",
+    "lint": "eslint --ext .js,.jsx .",
```

Comment:
This rule works for me now 🎉  

Static-site files are not linting in VS-Code for me (the linting of server-code works as expected). Perhaps this isn't possible as it'd require vs-code to be running multiple eslint versions? If it's working for you please ignore the following which must be specific to my setup & which we can debug elsewhere. 

Here's the eslint output (from within vs-code):
```
[Info  - 12:16:47] ESLint server is starting.
[Info  - 12:16:47] ESLint server running in node v18.15.0
[Info  - 12:16:47] ESLint server is running.
Uncaught exception received.
Error: spawn /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Plugin).app/Contents/MacOS/Code Helper (Plugin) ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:283:19)
    at onErrorNT (node:internal/child_process:476:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
[Info  - 12:16:47] ESLint library loaded from: /Users/naboo/github/nextstrain/nextstrain.org/node_modules/eslint/lib/api.js
[Info  - 12:16:47] ESLint library loaded from: /Users/naboo/github/nextstrain/nextstrain.org/static-site/node_modules/eslint/lib/api.js
```

Note: I had to test with this PR in combination with #754 so my package-lock isn't identical to this PR's.

Reply:
No problem, I think fine to debug in this thread. I am using VSCode extension [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) v2.4.2 and this is my output (I had to open a JS file in both static-site and server code to get the last 2 lines to show):

```
[Info  - 3:30:18 PM] ESLint server is starting.
[Info  - 3:30:19 PM] ESLint server running in node v18.15.0
[Info  - 3:30:19 PM] ESLint server is running.
[Info  - 3:30:19 PM] ESLint library loaded from: /Users/vlin/Dropbox/repos/nextstrain/nextstrain.org/static-site/node_modules/eslint/lib/api.js
[Info  - 3:30:19 PM] ESLint library loaded from: /Users/vlin/Dropbox/repos/nextstrain/nextstrain.org/node_modules/eslint/lib/api.js
```

What version of the extension do you have?

Reply:
I also have v2.4.2. Apart from the uncaught exception I'm experiencing the output looks the same. Does the linting actually work for you in the open JS files? As in, if you introduce an error do you get red-squiggly underlines & eslint on-hover infobox?

Reply:
That's weird. Linting on static-site works fine for me. For example, if I remove the bypass `react/no-unknown-property: off` and open `static-site/src/components/Datasets/dataset-map.jsx`, I see:

![static-site VSCode ESLint error](https://github.com/nextstrain/nextstrain.org/assets/13424970/5fd58b7c-76f0-4fef-af6c-eeb8e64cdc66)

Linting on server code also works fine. When adding a unused variable to server.js, I see:

![server VSCode ESLint error](https://github.com/nextstrain/nextstrain.org/assets/13424970/f63b920c-97fa-4b2b-b554-6ea360bc6471)

Reply:
Thanks for confirming! I'll do a bit more digging when I get the chance.

Reply:
I did some digging and think I understand what's going on, although it'd be interesting to know if you can replicate.

The code I was adding to test whether linting was working doesn't seem to be covered by the current `static-site/.eslintrc.yaml`. I was testing via adding things like:
```js
const unused = 1; // expect no-unused-vars error
const bbb = {key: un_used}; // expect 'un_used is not defined no-undef'
```
For me to get these errors in the static site I had to add `eslint:recommended` as an extension. I think this should be added to our config, but I'm interested to know if you get these kinds of errors flagged up as well? I'm guessing the answer is no because there are a few of these errors currently in the static-site code.

Adding that extension flags up 91 linting errors in the current code. Adding `browser: true` to `env` reduces this to 42. Most of them are `error  'require' is not defined` and `error  'exports' is not defined` as gatsby code uses this style. However there are also legitimate errors flagged up.

P.S. Without adding `eslint:recommended`, I _am_ getting linting working correctly (vs-code and cli) for specific JSX/hooks errors - for instance if I remove the `react/prop-types: off` rule then I get those errors flagged up. This was previously not working as expected, so I'll continue to monitor this - perhaps it was another vscode extension (I disabled them all & restarted to debug this), perhaps it was transient. I don't know.

Reply:
Oh right, I forgot about `eslint:recommended`. We should be using that just as we do in the server code. I'll add it.

Reply:
Added `eslint:recommended` in 875ff4b...6ed471e.